atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/encoding/TicketsCellEncodingStrategy.java

```diff
@@ -111,4 +115,57 @@ public Stream<byte[]> getRowSetCoveringTimestampRange(long fromInclusive, long t
                 toInclusive + rowsPerQuantum - ((toInclusive % partitioningQuantum) % rowsPerQuantum) - 1);
         return LongStream.rangeClosed(startRow, endRow).mapToObj(TicketsCellEncodingStrategy::rowToBytes);
     }
+
+    public CellRangeQuery getRangeQueryCoveringTimestampRange(long fromInclusive, long toInclusive) {
+        long startPartition = getPartition(fromInclusive);
+        long endPartition = getPartition(fromInclusive);
+        if (startPartition == endPartition) {
+            return getQueryForSingleDoublyBoundedPartition(startPartition, fromInclusive, toInclusive);
+        }
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
+
+    private CellRangeQuery getQueryForSingleDoublyBoundedPartition(
+            long partitionNumber, long fromInclusive, long toInclusive) {
+        ImmutableCellRangeQuery.Builder builder = ImmutableCellRangeQuery.builder();
+        for (int rowOffset = 0; rowOffset < rowsPerQuantum; rowOffset++) {
+            long firstColumnInPartition = getPartitionOffsetRow(fromInclusive) <= rowOffset
+                    ? getColumnIndex(fromInclusive)
+                    : getColumnIndex(fromInclusive) + 1;
+            long lastColumnInPartition = getPartitionOffsetRow(toInclusive) >= rowOffset
+                    ? getColumnIndex(toInclusive)
+                    : getColumnIndex(toInclusive) - 1;
+            if (firstColumnInPartition <= lastColumnInPartition) {
```

Comment:
this is a bit tricky to read, can we just break the loop if given`rowOffset` < row for startInclusive. 
We can also enforce that `fromInclusive` <= `toInclusive`